src/fee_bump_transaction.js

```diff
@@ -0,0 +1,255 @@
+  }
+
+  get networkPassphrase() {
+    return this._networkPassphrase;
+  }
+
+  set networkPassphrase(networkPassphrase) {
+    this.innerTransaction.networkPassphrase = networkPassphrase;
+    this._networkPassphrase = networkPassphrase;
+  }
+
+  /**
+   * Signs the transaction with the given {@link Keypair}.
+   * @param {...Keypair} keypairs Keypairs of signers
+   * @returns {void}
+   */
+  sign(...keypairs) {
+    const txHash = this.hash();
+    each(keypairs, (kp) => {
+      const sig = kp.signDecorated(txHash);
+      this.signatures.push(sig);
+    });
+  }
+
+  /**
+   * Signs a transaction with the given {@link Keypair}. Useful if someone sends
+   * you a transaction XDR for you to sign and return (see
+   * {@link FeeBumpTransaction#addSignature} for how that works).
+   *
+   * When you get a transaction XDR to sign....
+   * - Instantiate a `Transaction` object with the XDR
+   * - Use {@link Keypair} to generate a keypair object for your Stellar seed.
+   * - Run `getKeypairSignature` with that keypair
+   * - Send back the signature along with your publicKey (not your secret seed!)
+   *
+   * Example:
+   * ```javascript
+   * // `transactionXDR` is a string from the person generating the transaction
+   * const transaction = new FeeBumpTransaction(transactionXDR, networkPassphrase);
+   * const keypair = Keypair.fromSecret(myStellarSeed);
+   * return transaction.getKeypairSignature(keypair);
+   * ```
+   *
+   * @param {Keypair} keypair Keypair of signer
+   * @returns {string} Signature string
+   */
+  getKeypairSignature(keypair) {
+    return keypair.sign(this.hash()).toString('base64');
+  }
+
+  /**
+   * Add a signature to the transaction. Useful when a party wants to pre-sign
+   * a transaction but doesn't want to give access to their secret keys.
+   * This will also verify whether the signature is valid.
+   *
+   * Here's how you would use this feature to solicit multiple signatures.
+   * - Use `TransactionBuilder` to build a new transaction.
+   * - Make sure to set a long enough timeout on that transaction to give your
+   * signers enough time to sign!
+   * - Once you build the transaction, use `transaction.toXDR()` to get the
+   * base64-encoded XDR string.
+   * - _Warning!_ Once you've built this transaction, don't submit any other
+   * transactions onto your account! Doing so will invalidate this pre-compiled
+   * transaction!
+   * - Send this XDR string to your other parties. They can use the instructions
+   * for {@link FeeBumpTransaction#getKeypairSignature} to sign the transaction.
+   * - They should send you back their `publicKey` and the `signature` string
+   * from {@link FeeBumpTransaction#getKeypairSignature}, both of which you pass to
+   * this function.
+   *
+   * @param {string} publicKey The public key of the signer
+   * @param {string} signature The base64 value of the signature XDR
+   * @returns {TransactionBuilder}
+   */
+  addSignature(publicKey = '', signature = '') {
```

Comment:
there is some code which is duplicated between  `FeeBumpTransaction` and `Transaction`. is there is an easy and clean way to share common code between the 2 implementations?

Reply:
sure, let me explore this a little bit.

Reply:
fixed